platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/State.java

```diff
@@ -139,14 +139,23 @@ public void setSwirldState(final SwirldState state) {
         setChild(ChildIndices.SWIRLD_STATE, state);
     }
 
+    /**
+     * Immutable platform state is not supported by this class.
+     */
+    @NonNull
+    @Override
+    public PlatformStateAccessor getReadablePlatformState() {
+        throw new UnsupportedOperationException(
+                "com.swirlds.platform.state.State doesn't support read-only mode for PlatformState");
```

Comment:
This seems surprising.  Why wouldn't `State` allow a read-only platform state (even if backed by a writable platform state)?

Reply:
Yes, you're right. I'll change it